### DISCUSSION TITLE
Ansys Geometry Core Python Interface Error: GeometryExitedError on Using read_existing_design

### DISCUSSION THREAD

**Discussion by XIAO-GUO-commint** [ansys/pyansys-geometry#872] [Q&A] [answered]
from ansys.geometry.core import launch_modeler_with_spaceclaim

modeler_discovery = launch_modeler_with_spaceclaim()
design1=modeler_discovery.read_existing_design()
I am trying to use read_existing_design from the Ansys Geometry Core to get the object of an already opened file and perform operations on it. However, I keep encountering the following error: GeometryExitedError: Geometry service connection terminated: Exception was thrown by handler.

**Comment by RobPasMue** [comment]
HI @XIAO-GUO-commint - couple of questions:

1. What version of Ansys SpaceClaim are you using?
2. What version of PyAnsys Geometry are you using? (i.e. pip freeze --> search for ``ansys-geometry-core``)
3. Do you already have an open Ansys SpaceClaim session when you are trying to run the above?

**Reply by XIAO-GUO-commint** [reply]
1.I am using the 23R2 version of SpaceClaim.
2.The version of PyAnsys Geometry I am using is 0.4.2.
3.I am able to open SpaceClaim using modeler_discovery = launch_modeler_with_spaceclaim(), but I am unable to obtain the modeler object using modeler_discovery.read_existing_design(). Additionally, when using run_discovery_script_file, I am unsure what the script_args parameter should be. I have checked the original documentation but did not find instructions on how to proceed.
I hope you can inform me, thank you!

**Reply by RobPasMue** [reply]
Hi @XIAO-GUO-commint - so, couple of things.

23R2 is a best-effort compatibility with Ansys SpaceClaim and Ansys Discovery from the PyAnsys Geometry client. This means that full support is not at all available (in fact, it may throw back errors). Have a look at the following table: https://geometry.docs.pyansys.com/version/stable/getting_started/compatibility.html

Also, support from PyAnsys Geometry to 23R2 is mostly limited to version 0.3.X as you may have seen in the previous table.

My recommendation would be that you install version 0.3.2 --> https://pypi.org/project/ansys-geometry-core/0.3.2/

And try the previous operations. Let me know how it goes.

**Reply by XIAO-GUO-commint** [reply]
@RobPasMue Thank you for your response. I have already installed version 0.3.2, but I am still encountering two issues. 
(1) I cannot obtain the design object for an opened part.
![image](https://github.com/ansys/pyansys-geometry/assets/83817101/b99024be-1dad-4730-a936-eb59ff9e42a4)
I can only obtain a new design object using create_design instead of using an already-designed 3D model.
![image](https://github.com/ansys/pyansys-geometry/assets/83817101/ffb8425c-8edb-4f4d-8406-bd5279092c1b)
(2) I am unable to run run_discovery_script_file because I do not know how to fill in the script_args.
![image](https://github.com/ansys/pyansys-geometry/assets/83817101/323a422c-f5fb-4c9a-bbd0-51622ff06253)

**Reply by RobPasMue** [reply, marked as answer]
Regarding 2 - we solved that issue in new versions (your change is in ``main``). Please pass in an empty dictionary for now (i.e. ``{}``) if you do not need any script arguments

Regarding 1 - we may need a bit more information. In any case, I see that you are creating a ``modeler_discovery`` object. Are you trying to connect to Ansys Discovery? If this is the case, please check the table again. Version 23R2 of Ansys Discovery is not compatible with PyAnsys Geometry 0.3.X.

**Reply by XIAO-GUO-commint** [reply]
@RobPasMue Thank you for your reply. By using modeler_discovery = launch_modeler_with_spaceclaim(), I can connect to SpaceClaim. However, I cannot obtain a design object using design = modeler_discovery.read_existing_design() (likely due to compatibility issues). I can only obtain a design object using design = modeler_discovery.create_design('gem2'). My destination is simple: I just want to obtain a design object for an already-created 3D model.

Regarding your second question, you mentioned that in the new version, the issue can be resolved by passing an empty dictionary. Does that mean it can only be achieved in version 0.4.2?

**Reply by XIAO-GUO-commint** [reply]
Hello, I have one more question. Is the file_path parameter in run_discovery_script_file referring to an internally script file in SpaceClaim?
![image](https://github.com/ansys/pyansys-geometry/assets/83817101/1fd871d5-de60-43dd-8073-514c3fe4a327)
I have filled in the file_path with the path where the script is saved.

**Reply by RobPasMue** [reply]
Yes - it refers to an internal SpaceClaim script just like the one you were sharing in your screenshot 😄 - and the ``file_path`` arg should have as value the path to your local script file (just like you did)

**Comment by XIAO-GUO-commint** [comment]
![image](https://github.com/ansys/pyansys-geometry/assets/83817101/5fe307f2-9249-4d12-8d22-7f5a8c924d40)
I still have some problems...

**Comment by RobPasMue** [comment]
Hi @XIAO-GUO-commint - let me answer to your different points:

1. Versions ``0.4.X`` will be compatible with 24R1 onwards, and not 23R2. There are some compatibility breaking changes there (since 23R2 was on a best-effort basis)
2. The empty dictionary, you should be able to pass it in version 0.3.2, that's not a problem. What I meant is that following issue #870 we implemented a fix by which the default scenario is that no ``script_args`` are needed. But you will still be able to pass them if needed (it is just an optional argument now)
3. The 23R2 version has problems "reading" existing designs but not creating them from scratch from what I can see. Keep in mind that when you create a design, you are creating an empty SpaceClaim Document.
4. Whenever you call ``launch_modeler_with_spaceclaim()``  make sure that your previous SpaceClaim sessions are closed. Let PyAnsys Geometry open SpaceClaim for you.
5. Regarding your last error (https://github.com/ansys/pyansys-geometry/discussions/872#discussioncomment-7764086) - could you share your PyAnsys Geometry script with us (here in the discussion)?

**Reply by XIAO-GUO-commint** [reply]
@RobPasMue Thank you for your reply!
This is my code; the script file cannot be executed.
https://github.com/ansys/pyansys-geometry/assets/83817101/7e12a2f7-129e-4e16-8950-4e413cd6c966
![image](https://github.com/ansys/pyansys-geometry/assets/83817101/035d0f0c-7b5a-48ba-a23e-4f41019c6c60)

**Reply by RobPasMue** [reply]
Interesting... I saw the recording you shared. It seems like the script does get executed (I can see a 3D model right at the end of the video) but it might be throwing some error during the execution... SpaceClaim will generate some logs when using PyAnsys Geometry. On version 0.3.X these logs are not easily accessible but you should be able to see them at

C:\Users\[username]\AppData\Roaming\SpaceClaim\Log Files

Can you check them?

Also, you can increase the logging level of SpaceClaim from PyAnsys Geometry using the different args: https://geometry.docs.pyansys.com/version/0.3/api/ansys/geometry/core/connection/launcher/index.html#launcher.launch_modeler_with_discovery

 Check the previous link for the 0.3.X docs 😄

**Reply by XIAO-GUO-commint** [reply]
The 3D model at the end of the video is opened using modeler_discovery.open_file and not run_discovery_script_file, so the run_discovery_script_file function is still unusable (it might still be a compatibility issue). Will this issue be resolved in newer versions, such as 0.5.2 or fluent24r2?
[SpaceClaim_2023_12_5_2456.log](https://github.com/ansys/pyansys-geometry/files/13561556/SpaceClaim_2023_12_5_2456.log)
[SpaceClaim_2023_12_5_16124.log](https://github.com/ansys/pyansys-geometry/files/13561557/SpaceClaim_2023_12_5_16124.log)
[SpaceClaim_2023_12_5_23388.log](https://github.com/ansys/pyansys-geometry/files/13561558/SpaceClaim_2023_12_5_23388.log)
[SpaceClaim_2023_12_5_29820.log](https://github.com/ansys/pyansys-geometry/files/13561559/SpaceClaim_2023_12_5_29820.log)

**Reply by RobPasMue** [reply]
It might be a compatibility issue indeed... @b-matteo do you know if it is the case? (Ansys SpaceClaim 23R2 - run script file feature)

**Reply by XIAO-GUO-commint** [reply]
Thank you very much for your assistance; I hope the aforementioned issues can be resolved in future versions.

**Reply by RobPasMue** [reply]
So after confirming - 23R2 run_discovery_script_file feature with SpaceClaim is broken in PyAnsys Geometry 0.3.2. This has been solved for 24R1. Sorry for the inconveniences @XIAO-GUO-commint

**Reply by RobPasMue** [reply]
I will close the discussion. Thanks for all your interactions!
